plugins/catalog-backend/src/modules/codeowners/lib/resolve.ts

```diff
@@ -16,26 +16,34 @@
 
 import * as codeowners from 'codeowners-utils';
 import { CodeOwnersEntry } from 'codeowners-utils';
-import { filter, get, head, pipe, reverse } from 'lodash/fp';
+import { get, head, pipe, reverse } from 'lodash/fp';
 
 const USER_PATTERN = /^@.*/;
 const GROUP_PATTERN = /^@.*\/.*/;
 const EMAIL_PATTERN = /^.*@.*\..*$/;
 
 export function resolveCodeOwner(
   contents: string,
-  pattern = '*',
+  pattern?: string,
 ): string | undefined {
   const owners = codeowners.parse(contents);
 
-  return pipe(
-    filter((e: CodeOwnersEntry) => e.pattern === pattern),
+  let filteredOwners = owners.filter((e: CodeOwnersEntry) =>
+    pattern?.includes(e.pattern),
```

Comment:
`.includes` is much too weak, you'll need to

- parse the incoming URL (maybe using `git-url-parse` like other parts of the repo do?)
- extract the path part out of it
- make a proper check that the codeowners entry exactly equals, or that the path starts with `codeOwnersRowPattern + '/'` and possibly even handling backslash instead of slash. Otherwise you'd spuriously match a codeowners rule for `/catalog` with `/catalog-backend/catalog-info.yaml` which is not right

Reply:
I think it could get tricky since the codowners entries can have glob patterns, wildcards, etc. I do think using `git-url-parse` could help extract the path portion decently to help. I haven't thought of this, but perhaps there is some way to do an intersect or something on a glob pattern?

NOTE: This is partially one of the reasons, I didn't try to support monorepos in the first place do to the varying possibilities. That said, I'm fully supportive of adding it assuming it works well and handles most common usages properly.

Reply:
Yea I got aware that the codeowners can be pretty complex. I was thinking we could go for a restricted release where we support only the explicit use cases, for example.

Reply:
@andrewthauer I opted to use the codeowners.matchFile function from the codeowners-utils package to do the matching, it satisfies the tests and I think it makes sense to use the matching logic from that package. Do you remember why did you opt for doing the initial match on your own in this code ?

Reply:
@kissmikijr - TBH, I don't recall at this point. If it works that's great.